pkg/collector/corechecks/servicediscovery/service_detector.go

```diff
@@ -71,3 +74,18 @@ func (sd *serviceDetector) Detect(p processInfo) serviceMetadata {
 		FromDDService:      meta.FromDDService,
 	}
 }
+
+func ensureEnvWithPWD(env []string, cwd string) []string {
+	hasPWD := false
+	for _, v := range env {
+		n := strings.Split(v, "=")[0]
```

Comment:
could eliminate `hasPWD` entirely:

```
for _, v := range env {
    if strings.HasPrefix(v, "PWD=") {
        return env
    }
}
env = append(env, fmt.Sprintf("PWD=%s",cwd))
return env
```

Reply:
true, thanks for pointing this out! I just copied the previous implementation which wasn't in its own function and I missed this :D